Token detection failing

I get the following error message when I add markdown styling around the changelog entry pattern tokens.

> /home/runner/work/_temp/77e673da-ae35-48a4-ab38-6649cda9ba0e.sh: line 1: [old]: command not found
/home/runner/work/_temp/77e673da-ae35-48a4-ab38-6649cda9ba0e.sh: line 1: [new]: command not found
Error: Missing token: [old]
Error: Process completed with exit code 1.

This is my configuration.

```yaml
      - name: Changelog
        uses: crambl/dependabot-changelog-writer@b337c10b931c96fa5b44e6d2e25c1cd693c5eb36 # v1.1.0
        with:
          push-token: ${{ steps.sts.outputs.token }}
          commit-message: "chore: Update changelog for Dependabot"
          changelog: ./CHANGELOG.md
          section-header: "### Changes"
          changelog-entry-pattern: "Update _[dep]_ from `[old]` to `[new]`. ([pr-link]) _@dependabot_"
          push-changes: true
```

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -125,7 +125,17 @@ jobs:
           RUST_LOG: DEBUG
           GH_TOKEN: ${{ github.token }}
           PUSH_TOKEN: ${{ github.token }}
-        run: cargo run -- "CHANGELOG.md" "📝 Update [dep] from [old] to [new]" "Some commit message" "github-actions[bot]" "github-actions[bot]@users.noreply.github.com" "Unreleased" "section header 3" "false"
+        shell: bash
+        # Test for https://github.com/CramBL/dependabot-changelog-writer/issues/75
+        run: cargo run -- \
+            "CHANGELOG.md" \
+            'Update _[dep]_ from `[old]` to `[new]`. ([pr-link]) _@dependabot_' \
+            "Some commit message" \
+            "github-actions[bot]" \
+            "github-actions[bot]@users.noreply.github.com" \
+            "Unreleased" \
+            "section header 3" \
+            "false"
 
   build:
     strategy:
@@ -177,14 +187,14 @@ jobs:
           RUST_LOG: DEBUG
           GH_TOKEN: ${{ github.token }}
           PUSH_TOKEN: ${{ github.token }}
-        run: ./${{ env.binary_name }} "CHANGELOG.md" \
-                                    "📝 Update [dep] from [old] to🚀 [new]🍄"
-                                    "commit msg" \
-                                    "github-actions[bot]" \
-                                    "github-actions[bot]@users.noreply.github.com" \
-                                    "Unreleased" \
-                                    "section header 3" \
-                                    "false"
+        run: ./${{ env.binary_name }} 'CHANGELOG.md' \
+                                    '📝 Update [dep] from [old] to🚀 [new]🍄'
+                                    'commit msg' \
+                                    'github-actions[bot]' \
+                                    'github-actions[bot]@users.noreply.github.com' \
+                                    'Unreleased' \
+                                    'section header 3' \
+                                    'false'
 
   msrv:
       # check that we can build using the minimal rust version that is specified by this crate

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.1.1]
+
+### Fixed
+
+- Ensure inputs passed as arguments to the binary are not interpreted by bash before hand, fixes [#75](https://github.com/CramBL/dependabot-changelog-writer/issues/75)
+
 ## [1.1.0]
 
 ### Changed
@@ -57,13 +63,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `serde_json`: 1.0.137 → 1.0.138 ([#54](https://github.com/CramBL/dependabot-changelog-writer/pull/54))
 - `tempfile`: 3.15.0 → 3.16.0 ([#54](https://github.com/CramBL/dependabot-changelog-writer/pull/54))
 - `OpenSSL`: 3.4.0 → 3.4.1
-- `cargo update` 
+- `cargo update`
 
 ## [1.0.1]
 
 ### Changed
 
-- When `push-changes` was set to false, the would-be changelog diff was printed but the changelog was not actually changed, it is now. 
+- When `push-changes` was set to false, the would-be changelog diff was printed but the changelog was not actually changed, it is now.
 - When the actions runs, it now downloads to a uniquely named temporary directory, and cleans it up before the next step.
 
 ### Dependencies

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 name = "dependabot-changelog-writer"
 description = "Generate a changelog entry from a dependabot PR"
-version = "1.1.0"
+version = "1.1.1"
 edition = "2021"
 rust-version = "1.81"
 
@@ -21,7 +21,7 @@ testresult = "0.4.1"
 assert_cmd = "2.0.17"
 predicates = "3.1.3"
 
-# The binary is plenty fast, we optimize for size to reduce the amount that has to be downloaded when the action is run 
+# The binary is plenty fast, we optimize for size to reduce the amount that has to be downloaded when the action is run
 [profile.release]
 strip = true
 panic = "abort"

--- a/action.yml
+++ b/action.yml
@@ -95,8 +95,11 @@ runs:
           version="$(gh api -H "Accept: application/vnd.github+json" \
                             -H "X-GitHub-Api-Version: 2022-11-28" \
                             /repos/${action_repository}/git/refs/tags \
-                            | jq --raw-output --arg sha "${action_ref}" \
-                                '[.[] | select(.object.sha == $sha) | .ref | split("/")[-1]][-1] // ""')"
+                            | jq --raw-output --arg sha "${action_ref}" '[.[] \
+                                | select(.object.sha == $sha) \
+                                | .ref \
+                                | split("/")[-1]][-1] // ""' \
+                                )"
           if [[ -z "${version}" ]]; then
             echo "Could not find a tag with the digest ${action_ref}" >&2
             exit 1
@@ -115,7 +118,10 @@ runs:
       shell: bash
       run: |
         mkdir -p ${{ env.temp_dir }}  # Create a unique temporary directory
-        gh release download ${{ steps.get_action_version.outputs.version }} --repo ${{ env.action_org }}/${{ env.action_repo }} --pattern '${{ runner.os }}.tgz' --dir ${{ env.temp_dir }}
+        gh release download ${{ steps.get_action_version.outputs.version }} \
+            --repo ${{ env.action_org }}/${{ env.action_repo }} \
+            --pattern '${{ runner.os }}.tgz' \
+            --dir ${{ env.temp_dir }}
         tar -xzf ${{ env.temp_dir }}/${{ runner.os }}.tgz -C ${{ env.temp_dir }}
       env:
         GH_TOKEN: ${{ inputs.action-download-token }}
@@ -123,7 +129,15 @@ runs:
     - name: Run Action
       shell: bash
       id: run
-      run: ${{ env.temp_dir }}/${{ runner.os }}/${{ env.binary_name }} "${{ inputs.changelog }}" "${{ inputs.changelog-entry-pattern }}" "${{ inputs.commit-message }}" "${{ inputs.committer-name }}" "${{ inputs.committer-email }}" "${{ inputs.version }}" "${{ inputs.section-header }}" "${{ inputs.push-changes }}"
+      run: ${{ env.temp_dir }}/${{ runner.os }}/${{ env.binary_name }} \
+        '${{ inputs.changelog }}' \
+        '${{ inputs.changelog-entry-pattern }}' \
+        '${{ inputs.commit-message }}' \
+        '${{ inputs.committer-name }}' \
+        '${{ inputs.committer-email }}' \
+        '${{ inputs.version }}' \
+        '${{ inputs.section-header }}' \
+        '${{ inputs.push-changes }}'
       env:
         GH_TOKEN: ${{ inputs.action-download-token }}
         PUSH_TOKEN: ${{ inputs.push-token }}

